apps/css-workshop/src/components/Button.astro

```diff
@@ -5,20 +5,33 @@ type Props = {
   isActive?: boolean;
   dropdown?: boolean;
   idea?: boolean;
+  as?: string;
 } & astroHTML.JSX.ButtonHTMLAttributes;
 
-const { class: className, variant, size, isActive, dropdown, idea, ...props } = Astro.props;
+const {
+  as: Element = 'button',
+  class: className,
+  variant,
+  size,
+  isActive,
+  dropdown,
+  idea,
+  ...props
+} = Astro.props;
 ---
 
-<button
+<Element
   class:list={[
-    'iui-field iui-button',
-    { 'iui-button-dropdown': iuiType === 'dropdown' },
-    { 'iui-button-idea': iuiType === 'feedback' },
+    'iui-button',
+    'iui-button-base',
+    'iui-field',
+    { 'iui-button-dropdown': dropdown },
+    { 'iui-button-idea': idea },
     className,
   ]}
-  data-iui-variant={variant}
+  type='button'
```

Comment:
Should this be there only when `Element === "button"`

Reply:
We could do something like this:

```suggestion
  type={Element === 'button' ? 'button' : undefined}
```

But it's complicated. It wouldn't work for more advanced cases `as={SomeButton}`.

I'd say lets leave it as-is, since this isn't a public component.